pkg/clients/has/components.go

```diff
@@ -401,11 +406,19 @@ func (h *HasController) RetriggerComponentPipelineRun(component *appservice.Comp
 				return "", fmt.Errorf("cannot retrigger PipelineRun - required annotation %q not found", targetBranchAnnotationName)
 			}
 		}
-		file, err := h.Github.CreateFile(repoName, util.GenerateRandomString(5), "test", branchName)
-		if err != nil {
-			return "", fmt.Errorf("failed to retrigger PipelineRun %s in %s namespace: %+v", pr.GetName(), pr.GetNamespace(), err)
+
+		if gitProvider == "gitlab" {
+			_, err := h.GitLab.CreateFile(util.GenerateRandomString(5), "test", branchName)
+			if err != nil {
+				return "", fmt.Errorf("failed to retrigger PipelineRun %s in %s namespace: %+v", pr.GetName(), pr.GetNamespace(), err)
+			}
+		} else {
+			file, err := h.Github.CreateFile(repoName, util.GenerateRandomString(5), "test", branchName)
+			if err != nil {
+				return "", fmt.Errorf("failed to retrigger PipelineRun %s in %s namespace: %+v", pr.GetName(), pr.GetNamespace(), err)
+			}
+			sha = file.GetSHA()
 		}
-		sha = file.GetSHA()
```

Comment:
shouldn't we update SHA for gitlab PLR as well?

Reply:
good question, the [FileInfo struct](https://pkg.go.dev/github.com/xanzy/go-gitlab@v0.104.1#FileInfo) doesn't contain the sha of the created file unfortunately.
Do you think it's worth getting the sha anyway? There might be another way to get the sha by some other API calls.

Reply:
ok I see that it's currently not relevant for the test, because it gets the current SHA [later in the test](https://github.com/konflux-ci/e2e-tests/blob/7b3e0e92618b32ab1b4fe6cb9eae519586e9b994/tests/integration-service/gitlab-integration-reporting.go#L162-L168), so i'm ok with keeping it as is

Reply:
makes sense, thanks